core/modules/replica/ConfigurationSchema.h

```diff
@@ -0,0 +1,206 @@
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+
+// Forward declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+    class Configuration;
+namespace database {
+namespace mysql {
+    class Connection;
+}}  // namespace database::mysql
+}}} // namespace lsst::qserv::replica
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * This utility class ConfigurationSchema encaposulates the JSON and MySQL schemas
+ * and operations over the schemas of the Configuration service. Both schemas are
+ * put into this class to facilitate schema changes tracking, unit and integration
+ * testing, as well as other actions that may involve the schemas, such as schema
+ * evolution, schema verification, etc.
+ *
+ * @note This class is the primary source of the schemas for the given version.
+ */
+class ConfigurationSchema {
+public:
+    /// The current version number required by the application.
+    static int const version;
+
+    /// @return A documentation string for the specified parameter or the empty string
+    ///   if none is available in the schema.
+    static std::string description(std::string const& category, std::string const& param);
+
+    /// @return A 'true' if the parameter can't be modified via the 'set' methods
+    ///   of the Configuration class. This information is used by class Configuration
+    ///   to validate the parameters.
+    static bool readOnly(std::string const& category, std::string const& param);
+
+    /// @return A 'true' if the parameter represents the security context (passwords,
+    ///   authorization keys, etc.). Parameters possesing this attribute are supposed
+    ///   to be used with care by the dependent automation tools to avoid exposing
+    ///   sensitive information in log files, reports, etc.
+    static bool securityContext(std::string const& category, std::string const& param);
+
+    /// @return A 'true' if, depending on the actual type of the parameter, the empty
+    ///   string (for strings) or zero value (for numeric parameters) is allowed.
+    ///   This information is used by class Configuration to validate input values
+    ///   of the parameters.
+    static bool emptyAllowed(std::string const& category, std::string const& param);
+
+    /// @return The default configuration data as per the current JSON schema to be loaded
+    ///   into the transent state of the class Configuration upon its initialization.
+    static nlohmann::json defaultConfigData();
+
+    /// @return The configuration data for the unit testing. The data is compatible with
+    ///   the current JSON configuration schema. In addition to the overwritten default
+    ///   of the general parameters it also containers tst definitions for the group data,
+    ///   that includes workers, database families and databases.
+    static nlohmann::json testConfigData();
```

Comment:
Initially, I had the test object within the unit test application `testConfiguration.cc` which is the only client of this method. That unit test is already huge, so I later decided to move the test close to where the default schema is so that it all would be kept at one location. The latter might be helpful should any new parameters be added/removed in the future. 
Though, I will seriously consider your suggestion.